app/cmd/snapshot.go

```diff
@@ -326,3 +365,52 @@ func infoSnapshot(c *cli.Context) error {
 	fmt.Println(string(output))
 	return nil
 }
+
+func cloneSnapshot(c *cli.Context) error {
+	snapshotName := c.String("snapshot-name")
+	if snapshotName == "" {
+		return fmt.Errorf("missing required parameter --snapshot-name")
+	}
+	fromControllerAddress := c.String("from-controller-address")
+	if fromControllerAddress == "" {
+		return fmt.Errorf("missing required parameter --from-controller-address")
+	}
+	exportBackingImageIfExist := c.Bool("export-backing-image-if-exist")
+
+	controllerClient, err := getControllerClient(c)
+	if err != nil {
+		return err
+	}
+	defer controllerClient.Close()
+
+	fromControllerClient, err := client.NewControllerClient(fromControllerAddress)
```

Comment:
looks like this client is leaked?

You should be able to close it in a defer here, since the following code seems to be blocking `defer fromControllerClient.Close()`

Reply:
Thanks Joshua for the catch! Yeah, I missed this one. Fixed